includes/admin/class-wc-rest-payments-orders-controller.php

```diff
@@ -199,4 +208,49 @@ public function create_customer( $request ) {
 			return new WP_Error( 'wcpay_server_error', __( 'Unexpected server error', 'woocommerce-payments' ), [ 'status' => 500 ] );
 		}
 	}
+
+	/**
+	 * Create a new in-person payment intent for the given order ID without confirming it.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function create_terminal_intent( $request ) {
+		try {
+			$order_id        = $request['order_id'];
+			$payment_methods = [ 'card_present' ];
+			$capture_method  = 'manual';
+
+			// Do not process non-existing orders.
+			$order = wc_get_order( $order_id );
+			if ( false === $order ) {
+				return new WP_Error( 'wcpay_missing_order', __( 'Order not found', 'woocommerce-payments' ), [ 'status' => 404 ] );
+			}
+
+			// Create a new intention.
+			$result = $this->gateway->create_intent( $order, $payment_methods, $capture_method );
+			if ( 'created' !== $result['status'] ) {
```

Comment:
in order to avoid adding new status variable with different values, we could catch `API_Exception` here.

Reply:
I'm not sure if I follow. Could you give an example?

Reply:
I mean you can let `create_intent` to throw `API_Exception` ( now you catch the exception inside the method) and catch it here . this way you can avoid introducing status 'created'/'failed'.

Reply:
I've refactored it as per this comment:  
https://github.com/Automattic/woocommerce-payments/pull/3465#discussion_r763068901

In short, `create_intent` is still catching `API_Exception` but rather than returning a convoluted response array it's not returning a `WP_Error` object on failure and an intent id on success.